jobs/dtd-metrics/src/services/global-store.service.ts

```diff
@@ -0,0 +1,91 @@
+import { Attribute, ReadModelClient } from '@interop-be-reports/commons'
+import { MACRO_CATEGORIES } from '../configs/macro-categories.js'
+import { MacroCategories, MacroCategory } from '../models/macro-categories.model.js'
+import { z } from 'zod'
+
+const GlobalStoreTenant = z.object({
+  id: z.string(),
+  name: z.string(),
+  createdAt: z.coerce.date(),
+  selfcareId: z.string().optional(),
+  macroCategoryId: z.string(),
+})
+type GlobalStoreTenant = z.infer<typeof GlobalStoreTenant>
+
+export class GlobalStoreService {
+  tenants: Array<GlobalStoreTenant>
+  onboardedTenants: Array<GlobalStoreTenant>
+  tenantsMap: Map<string, GlobalStoreTenant>
+  macroCategories: MacroCategories
+
+  public getMacroCategoryFromTenantId(tenantId: string): MacroCategory | undefined {
+    return this.macroCategories.find(({ tenantsIds }) => tenantsIds.has(tenantId))
+  }
+
+  public getTenantFromId(tenantId: string): GlobalStoreTenant | undefined {
+    return this.tenantsMap.get(tenantId)
+  }
+
+  private constructor(tenants: Array<GlobalStoreTenant>, macroCategories: MacroCategories) {
+    this.tenants = tenants
+    this.onboardedTenants = tenants.filter(({ selfcareId }) => !!selfcareId)
+    this.tenantsMap = new Map(tenants.map((tenant) => [tenant.id, tenant]))
+    this.macroCategories = macroCategories
+  }
+
+  static async init(readModel: ReadModelClient): Promise<GlobalStoreService> {
+    const attributes = await readModel.attributes
+      .find({
+        'data.code': {
+          $in: MACRO_CATEGORIES.flatMap((macroCategory) => macroCategory.ipaCodes),
+        },
+      })
+      .project({
+        _id: 0,
+        'data.id': 1,
+        'data.code': 1,
+      })
+      .map(({ data }) => Attribute.pick({ id: true, code: true }).parse(data))
+      .toArray()
+
+    const enrichMacroCategory = async (macroCategory: (typeof MACRO_CATEGORIES)[number]): Promise<MacroCategory> => {
+      const macroCategoryAttributes = attributes
+        .filter(({ code }) => (macroCategory.ipaCodes as ReadonlyArray<string | undefined>).includes(code))
+        .map((attribute) => ({ ...attribute, macroCategoryId: macroCategory.id }))
+
+      const macroCategoryTenants = await readModel.tenants
+        .find(
```

Comment:
May I suggest limiting this query only to tenants with the selfcareId?

Reply:
That would save us a lot of mongo computation, but we have also a metrics that also need to count not onboarded tenants

Reply:
a note for the future: consider that non-onboarded tenants will probably be removed (because the only source will be onboards made on selfcare)

Reply:
Uhmmm ok, but this metric exists only for a fluke.
At the moment we are loading all IPA institutions, but this feature could be dropped.
Anyway, let's merge the PR.
cc @ruggerocastagnola